src/components/plots/FrequencyPlot.jsx

```diff
@@ -27,11 +26,7 @@ const FrequencyPlot = (props) => {
     }
   }, []);
 
-  useEffect(() => {
-    if (hierarchy && properties) {
-      setPlotSpec(generateSpec(config, generateData(hierarchy, properties, config)));
-    }
-  }, [hierarchy, properties]);
+  const plotSpec = generateSpec(config, generateData(hierarchy, properties, config));
```

Comment:
Instead of this, could you make `useEffect` dependent on `hierarchy`, `properties`, and `config` (i.e. adding `config` as a variable to the original useEffect)? This would make it much faster if there were other variables that can cause this to rerender in the future.